### PR TITLE
py-maestrowf: Addition of new releases.

### PR DESCRIPTION
**Pull request by FrankD412**

### CODE REVIEW

File: var/spack/repos/builtin/packages/py-maestrowf/package.py [hunk 1]
```diff
@@ -30,11 +30,16 @@ class PyMaestrowf(PythonPackage):
...
+    version('1.1.1', 'd38bbf634de4f29fd01d1864ba2f70e0')
+    version('1.1.0', '3c20bf36fbb85d14c3bfdb865944a409')
     version('1.0.1', '6838fc8bdc7ca0c1adbb6a0333f005b4')
 
     depends_on('py-setuptools', type='build')
     depends_on('py-pyyaml',     type=('build', 'run'))
     depends_on('py-six',        type=('build', 'run'))
     depends_on('py-enum34',     type=('build', 'run'))
+    depends_on('py-tabulate',   type=('build', 'run'))
+    depends_on('py-filelock',   type=('build', 'run'))
```
Comment (healther) [inline var/spack/repos/builtin/packages/py-maestrowf/package.py]: are these requirements for all versions or just for the new ones?

Comment (FrankD412) [reply var/spack/repos/builtin/packages/py-maestrowf/package.py]: The newly added requirements are for newer versions. It looks like `depends_on` takes a `when` clause, though I haven't found a good example of the clause specifying dependencies when a version is equal to or greater than a particular release.

Comment (healther) [reply var/spack/repos/builtin/packages/py-maestrowf/package.py]: If you grep through the packages you should find that, in essence `when` takes a string that contains combinations of version (ranges) `@a.b.c:d.e.f`, variants `+variant`/`~variant` and other options like `arch=darwin` that restricts whatever attribute that it is added to.

In this case you end up with `depends_on('py-tabulate', type=..., when='@1.1.0:')`

Comment (FrankD412) [reply var/spack/repos/builtin/packages/py-maestrowf/package.py]: Got it -- corrected. Thanks for the info.

File: var/spack/repos/builtin/packages/py-maestrowf/package.py [hunk 2]
```diff
@@ -30,11 +30,16 @@ class PyMaestrowf(PythonPackage):
        simulation studies."""
 
     homepage = "https://github.com/LLNL/maestrowf/"
-    url      = "https://github.com/LLNL/maestrowf/archive/v1.0.1.tar.gz"
+    url      = "https://github.com/LLNL/maestrowf/archive/v1.1.2x.tar.gz"
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-maestrowf/package.py]: What's with the `x` at the end?

Comment (FrankD412) [reply var/spack/repos/builtin/packages/py-maestrowf/package.py]: You know, that's a good question. It appears like it might be a typo.

Comment (FrankD412) [reply var/spack/repos/builtin/packages/py-maestrowf/package.py]: @adamjstewart -- corrected.

**Pull request merged**